mephisto/abstractions/providers/prolific/prolific_unit.py

```diff
@@ -0,0 +1,365 @@
+
+    def register_from_provider_data(
+        self, prolific_study_id: str, prolific_submission_id: str,
+    ) -> None:
+        """Update the datastore and local information from this registration"""
+        self.datastore.register_submission_to_study(
+            prolific_study_id=prolific_study_id,
+            unit_id=self.db_id,
+            prolific_submission_id=prolific_submission_id,
+        )
+        self._sync_study_mapping()
+
+    def get_prolific_submission_id(self) -> Optional[str]:
+        """Return the Prolific Submission ID (assignment ID) associated with this unit"""
+        self._sync_study_mapping()
+        return self.prolific_submission_id
+
+    def get_prolific_study_id(self) -> Optional[str]:
+        """Return the Porlific Study ID associated with this unit"""
+        self._sync_study_mapping()
+        return self.prolific_study_id
+
+    def get_requester(self) -> 'ProlificRequester':
+        """Wrapper around regular Requester as this will be ProlificRequester"""
+        if self.__requester is None:
+            self.__requester = cast('ProlificRequester', super().get_requester())
+        return self.__requester
+
+    def get_status(self) -> str:
+        """Get status for this unit directly from Prolific, fall back on local info"""
+
+        # -------------------------------------------------------------------------
+        # TODO (#1008): Fix this logic. It looked abstract and unclear in Mturk code.
+        if self.db_status == AssignmentState.CREATED:
+            return super().get_status()
+        elif self.db_status in [
+            AssignmentState.ACCEPTED,
+            AssignmentState.EXPIRED,
+            AssignmentState.SOFT_REJECTED,
+        ]:
+            # These statuses don't change with a get_status call
+            return self.db_status
+
+        # -------------------------------------------------------------------------
+        # TODO (#1008): Fix this logic. It looked abstract and unclear in Mturk code.
+        if self.db_status in [AssignmentState.COMPLETED, AssignmentState.REJECTED]:
+            # These statuses only change on agent dependent changes
+            agent = self.get_assigned_agent()
+            found_status = self.db_status
+
+            if agent is not None:
+                agent_status = agent.get_status()
+                if agent_status == AgentState.STATUS_APPROVED:
+                    found_status = AssignmentState.ACCEPTED
+                elif agent_status == AgentState.STATUS_REJECTED:
+                    found_status = AssignmentState.REJECTED
+                elif agent_status == AgentState.STATUS_SOFT_REJECTED:
+                    found_status = AssignmentState.SOFT_REJECTED
+            else:
+                logger.warning(f'Agent for unit {self} is None')
+
+            if found_status != self.db_status:
+                self.set_db_status(found_status)
+
+            return self.db_status
```

Comment:
These are statuses where the status of the unit is determined by what we determine the agent status. Essentially, after a task is no longer live it's units are all in one of:
`[AssignmentState.ACCEPTED, AssignmentState.EXPIRED, AssignmentState.SOFT_REJECTED, AssignmentState.COMPLETED, AssignmentState.REJECTED]`
You can think of these roughly as `ACCEPTED` and `SOFT_REJECTED` corresponding with a "paid complete" status, which cannot be changed (can't unpay!), `COMPLETED` and `REJECTED` as "unpaid complete" (wherein payment may still occur, can reverse rejections in MTurk at least), and `EXPIRED` as "unpaid incomplete" (but can not be paid out, so it's final).

As the status for `COMPLETED` and `REJECTED` may change if we approve an agent, we need to check with the agent.